Scala/Scala.sublime-syntax

```diff
@@ -194,71 +370,97 @@ contexts:
+        - match: '\b(val)\b'
+          scope: storage.type.scala
+        - match: '({{alphaid}})(?=\s*:)'
           captures:
             1: variable.parameter.scala
+          push:
+            - match: '{{typeprefix}}'
+              set:
+                - match: '(?=\))'
+                  pop: true
+                - match: ','
+                  scope: punctuation.separator.scala
+                  pop: true
+                - include: delimited-type-expression
+                - match: '(?=[=])'
+                  pop: true
         - include: main
     - match: '(?=([\{\n]|extends))'
       pop: true
-  empty-parentheses:
-    - match: \(\)
-      scope: meta.parentheses.scala
+
   imports:
-    - match: \b(import)\s+
-      captures:
-        1: keyword.other.import.scala
+    - match: \b(import)\b
+      scope: keyword.other.import.scala
       push:
         - meta_scope: meta.import.scala
-        - match: '(?<=[\n;])'
+        - match: '(?=[\n;])'
           pop: true
-        - include: comments
-        - match: '([^\s{;.]+)\s*\.\s*'
+        - match: '(?:{{id}}\.)+{{id}}?'
           scope: variable.package.scala
-        - match: '([^\s{;.]+)\s*'
+        - match: '{{id}}'
           scope: variable.import.scala
+        - match: '_'
+          scope: variable.language.underscore.scala
         - match: "{"
           push:
             - meta_scope: meta.import.selector.scala
             - match: "}"
               pop: true
-            - match: |-
-                (?x) \s*
-                				([^\s.,}]+) \s*
-                				(=>) \s*
-                				([^\s.,}]+) \s*
-
+            - match: '({{id}})\s*({{rightarrow}})\s*({{id}})'
               captures:
                 1: variable.import.renamed-from.scala
-                2: keyword.other.arrow.scala
+                2: keyword.operator.arrow.scala
                 3: variable.import.renamed-to.scala
-            - match: '([^\s.,}]+)'
+            - match: '{{rightarrow}}'
+              scope: keyword.operator.arrow.scala
+            - match: '{{id}}'
               scope: variable.import.scala
+            - match: '_'
+              scope: variable.language.underscore.scala
+
   inheritance:
-    - match: '(extends|with)\s+([^\s\{\(\[\]]+)'
+    - match: '\b(extends|with)\s+([^\s\{\(\[\]]+)'
       captures:
         1: keyword.declaration.scala
         2: entity.other.inherited-class.scala
+    - match: '\b(extends|with)\s+\('
+      captures:
+        1: keyword.declaration.scala
+      push:
+        - match: '\)'
+          pop: true
+        - include: delimited-type-expression
+
+  # we need to greedily capture operators to prevent (e.g.) :: being matched as an ascription
+  # it would be incorrect to scope operators specially, since they are just identifiers
+  # by pulling them out HERE though and refusing to scope them, we emulate lookbehind
+  operators:
+    - match: '{{op}}'   # no explicit scope, just pulling it out
```

Comment:
Well, they aren't keywords any more than `JFrame` is a keyword in Java.  They're just identifiers like any other.  The only reason that I'm matching them *at all* is to ensure that I can match the primitive operators (the ones that *are* keywords) later on, since the primitive operators are valid substrings of non-primitive operators (and should not be matched in that case).  It's basically a backwards way of emulating word boundaries for symbolic identifiers.